File: benchmark/results/python-multilang/run-002/b-baseline/output/metaengine_demo/catalog/services/product_service.py
"""ProductService for the catalog domain."""

from typing import Optional

from metaengine_demo.catalog.aggregates.product import Product


class ProductService:
    """Application service exposing product operations."""

    def create(self, input: Optional[Product]) -> Product:
        """Create a new product from a partial input."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Product]:
        """Look up a product by its identifier."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Product]:
        """List products up to the given limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Delete a product by its identifier."""
        raise NotImplementedError("not implemented")

File: benchmark/results/ts-invocation-heredoc/run-003/b-baseline/output/src/domain/catalog/services/ProductService.ts
import { Product } from "../aggregates/Product";

/** Service for managing products. */
export class ProductService {
  create(input: Partial<Product>): Product {
    void input;
    throw new Error("not implemented");
  }

  findById(id: string): Product | null {
    void id;
    throw new Error("not implemented");
  }

  list(limit: number): Product[] {
    void limit;
    throw new Error("not implemented");
  }

  delete(id: string): void {
    void id;
    throw new Error("not implemented");
  }
}
